cmd/distributor/main.go

```diff
@@ -48,6 +48,8 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error initializing distributor: %v", err)
 	}
+	go dist.Run()
```

Comment:
Yeah stick it in the constructor please - and the rule should be changed too (https://github.com/weaveworks/cortex/blob/master/cmd/ruler/main.go#L44) - will file an issue for it.